homeassistant/components/octoprint/sensor.py

```diff
@@ -187,7 +187,9 @@ def native_value(self) -> datetime | None:
 
         read_time = self.coordinator.data["last_read_time"]
 
-        return read_time + timedelta(seconds=job.progress.print_time_left)
+        return (read_time + timedelta(seconds=job.progress.print_time_left)).replace(
+            second=0, microsecond=0
+        )
```

Comment:
We should not have to worry about the microseconds, as the timespec is already seconds in Home Assistant

```suggestion
        return (read_time + timedelta(seconds=job.progress.print_time_left)).replace(
            second=0
        )
```

Reply:
Ok, so even though `homeassistant.util.dt.utcnow()` returns a timestamp _with_ microseconds, the microseconds part is ignored when the sensor is updated?

Reply:
I don't have anything to print and test the change with at the moment, but the tests are green, so I suppose that's the way it works 👍

Reply:
> returns a timestamp with microseconds, the microseconds part is ignored when the sensor is updated?

The eventually returned/processed timestamp that is coming out of the sensor has a timespec of `seconds`, meaning microseconds are already disregarded.